### PR TITLE
REGR: undocumented astype("category").astype(str) type inconsistency between pandas 1.1 & 1.2

### LINKED ISSUE #41797: BUG: undocumented astype("category").astype(str) type inconsistency between pandas 1.1 & 1.2
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import numpy as np
import pandas as pd

print(pd.__version__)
df = pd.DataFrame(["a", "b", np.nan])
print("astype(str): \n", df.astype(str))
print("astype('category').astype(str): \n", df.astype("category").astype(str))
```

Output 1.1.3:
```
1.1.3
astype(str): 
      0
0    a
1    b
2  nan
astype('category').astype(str): 
      0
0    a
1    b
2  nan
```

Output 1.2.4:
```
1.2.4
astype(str): 
      0
0    a
1    b
2  nan
astype('category').astype(str): 
      0
0    a
1    b
2  NaN
```

#### Problem description

The above snippet run on version 1.1 (tested on 1.1.3) returns `'nan'` strings of the converted categories, whereas on 1.2 (tested on 1.2.4) keeps `np.nan` objects. Tangentially, when using the pandas StringDtype conversion (`.astype("string")`, pandas 1.1 and 1.2 both keep a NAType after conversion from categorical to string.

It is not clear whether it is expected behaviour that this conversion keeps `np.nan`, or whether it should return a `'nan'` string. The former would mean that this is now correct, the latter would mean there is a regression.

This issue is probably related to https://github.com/pandas-dev/pandas/pull/37355 and https://github.com/pandas-dev/pandas/issues/25353

#### Expected Output

Document what behaviour is desired.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.0-14-amd64
Version          : #1 SMP Debian 4.19.171-2 (2021-01-30)
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.4
numpy            : 1.20.2
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.1.2
setuptools       : 54.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.22.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 0.8.7
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 4.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : 0.53.1

</details>

Versions are identical except for one run at pandas 1.1.3, the other at 1.2.4.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #41797
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — APPROVED** [review]

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -523,14 +523,17 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
             try:
                 new_cats = np.asarray(self.categories)
                 new_cats = new_cats.astype(dtype=dtype, copy=copy)
+                fill_value = np.array(np.nan).astype(dtype).item()
```
Comment (jbrockmendel) [inline pandas/core/arrays/categorical.py:526]: there are cases where .item() has weird behavior and we use lib.item_from_zerodim instead.  no idea if any of those cases are reachable here

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Comment by jreback** [comment]
cc @simonjayhawkins attempted backport to 1.2.x & 1.3.x

**Comment by simonjayhawkins** [comment]
> cc @simonjayhawkins attempted backport to 1.2.x & 1.3.x

1.3.x backport ok, 1.2.x needs manual. will do shortly.

**Pull request merged**